x-pack/plugins/ml/public/application/components/annotations/annotation_flyout/index.test.tsx

```diff
@@ -5,58 +5,96 @@
     const annotation = mockAnnotations[1] as Annotation;
-    annotation$.next(annotation);
 
-    // useObservable wraps the observable in a new component
-    const ObservableComponent = (props: any) => {
-      const annotationProp = useObservable(annotation$);
-      if (annotationProp === undefined) {
-        return null;
-      }
-      return <AnnotationFlyout annotation={annotationProp} {...props} />;
-    };
-
-    const wrapper = mountWithIntl(<ObservableComponent />);
-    const updateBtn = wrapper.find('EuiButton').first();
-    expect(updateBtn.prop('isDisabled')).toEqual(true);
+    annotationUpdatesService.setValue(annotation);
+
+    render(
+      <MlAnnotationUpdatesContextProvider>
+        <ObservableComponent />
+      </MlAnnotationUpdatesContextProvider>
+    );
+    const updateBtn = screen.getByText('Update').closest('button');
+    expect(updateBtn).toBeDisabled();
   });
 
-  test('Error displayed and update button displayed if annotation text is longer than max chars', () => {
+  test('Error displayed and update button displayed if annotation text is longer than max chars', async () => {
     const annotation = mockAnnotations[2] as Annotation;
-    annotation$.next(annotation);
+    annotationUpdatesService.setValue(annotation);
 
     // useObservable wraps the observable in a new component
-    const ObservableComponent = (props: any) => {
-      const annotationProp = useObservable(annotation$);
-      if (annotationProp === undefined) {
-        return null;
-      }
-      return <AnnotationFlyout annotation={annotationProp} {...props} />;
-    };
-
-    const wrapper = mountWithIntl(<ObservableComponent />);
-    const updateBtn = wrapper.find('EuiButton').first();
-    expect(updateBtn.prop('isDisabled')).toEqual(true);
-
-    expect(wrapper.find('EuiFormErrorText')).toHaveLength(1);
+
+    render(
+      <MlAnnotationUpdatesContextProvider>
+        <ObservableComponent />
+      </MlAnnotationUpdatesContextProvider>
+    );
+    const updateBtn = screen.getByText('Update').closest('button');
+    expect(updateBtn).toBeDisabled();
+    await waitFor(() => {
+      const errorText = screen.queryByText(/characters above maximum length/);
+      expect(errorText).not.toBe(undefined);
+    });
+  });
+
+  test('Flyout disappears when annotation is updated', async () => {
+    const annotation = mockAnnotations[0] as Annotation;
+
+    annotationUpdatesService.setValue(annotation);
+
+    const { getByTestId } = render(
+      <MlAnnotationUpdatesContextProvider>
+        <ObservableComponent />
+      </MlAnnotationUpdatesContextProvider>
+    );
+    const updateBtn = getByTestId('annotationFlyoutUpdateButton');
+    expect(updateBtn).not.toBeDisabled();
+    expect(screen.queryByTestId('mlAnnotationFlyout')).toBeInTheDocument();
+
+    await fireEvent.click(updateBtn);
+    expect(screen.queryByTestId('mlAnnotationFlyout')).not.toBeInTheDocument();
+    screen.debug();
```

Comment:
do we need this?

Reply:
Removed here https://github.com/elastic/kibana/pull/83216/commits/1365c52b913b0c1fd556f5642d8ba74d0746c302